public/services/shared-link.ts

```diff
@@ -30,70 +31,86 @@ export async function addTenantToShareURL(core: CoreStart) {
     return;
   }
-  // Add the tenant to URLs copied from the share panel
+
   document.addEventListener('copy', (event) => {
-    const shareButton = document.querySelector('[data-share-url]');
-    const target = document.querySelector('body > span');
-    // The copy event listens to Cmd + C too, so we need to make sure
-    // that we're actually copied something via the share panel
-    if (
-      shareButton &&
-      target &&
-      shareButton.getAttribute('data-share-url') === target.textContent
-    ) {
-      const originalValue = target.textContent;
-      let urlPart = originalValue;
-
-      // We need to figure out where in the value to add the tenant.
-      // Since OpenSearchDashboards sometimes adds values that aren't in the current location/url,
-      // we need to use the actual input values to do a sanity check.
-      try {
-        // For the iFrame urls we need to parse out the src
-        if (originalValue && originalValue.toLowerCase().indexOf('<iframe') === 0) {
-          const regex = /<iframe[^>]*src="([^"]*)"/i;
-          const match = regex.exec(originalValue);
-          if (match) {
-            urlPart = match[1]; // Contains the matched src, [0] contains the string where the match was found
-          }
-        }
+    processCopyEvent(tenant);
+  });
+}
 
-        const newValue = addTenantToURL(urlPart!, originalValue!, tenant);
+export function processCopyEvent(userRequestedTenant: string) {
+  const shareButton = document.querySelector('[data-share-url]') as any;
+  const target = document.querySelector('body > span');
 
-        if (newValue !== originalValue) {
-          target.textContent = newValue;
+  // The copy event listens to Cmd + C too, so we need to make sure
+  // that we're actually copied something via the share panel
+  if (shareButton && target && shareButton.getAttribute('data-share-url') === target.textContent) {
+    const originalValue = target.textContent;
+    let urlPart = originalValue;
+
+    // We need to figure out where in the value to add the tenant.
+    // Since OpenSearchDashboards sometimes adds values that aren't in the current location/url,
+    // we need to use the actual input values to do a sanity check.
+    try {
+      // For the iFrame urls we need to parse out the src
+      if (originalValue && originalValue.toLowerCase().indexOf('<iframe') === 0) {
+        const regex = /<iframe[^>]*src="([^"]*)"/i;
+        const match = regex.exec(originalValue);
+        if (match) {
+          urlPart = match[1]; // Contains the matched src, [0] contains the string where the match was found
         }
-      } catch (error) {
-        // Probably wasn't an url, so we just ignore this
       }
+
+      updateClipboard(urlPart, originalValue, userRequestedTenant);
+    } catch (error) {
+      // Probably wasn't an url, so we just ignore this
     }
-  });
+  }
 }
 
-function addTenantToURL(
-  url: string,
+export function updateClipboard(
+  urlPart: string,
```

Comment:
What is the difference between urlPart and originalValue? From the `processCopyEvent` function it looks like urlPart is the originalValue unless originalValue is an `iframe`. If `originalValue` is an `iframe` then processCopyEvent will do some addition work to extract the `src`. Since the function to `updateClipboard` can sometimes return `originalValue` is there ever a case where it can return the `iframe` instead of the extracted `src`?

Reply:
Ignore this comment. I just checked the code on `main` and see how its related to the `addTenantToURL` method.